modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/SCIM2RestClient.java

```diff
@@ -880,6 +880,128 @@ public int attemptUserUpdate(PatchOperationRequestObject patchUserInfo, String u
         }
     }
 
+    /**
+     * Create a user with the provided bearer token.
+     *
+     * @param appRegisteredUserInfo UserObject containing user details.
+     * @param token                 Bearer token for authentication.
+     * @return ID of the created user.
+     */
+    public String createUserWithBearerToken(UserObject appRegisteredUserInfo, String token) {
+        String jsonRequest = toJSONString(appRegisteredUserInfo);
+        if (appRegisteredUserInfo.getScimSchemaExtensionEnterprise() != null) {
+            jsonRequest = jsonRequest.replace(SCIM_SCHEMA_EXTENSION_ENTERPRISE, USER_ENTERPRISE_SCHEMA);
+        }
+        if (appRegisteredUserInfo.getScimSchemaExtensionSystem() != null) {
+            jsonRequest = jsonRequest.replace(SCIM_SCHEMA_EXTENSION_SYSTEM, USER_SYSTEM_SCHEMA);
+        }
+
+        try (CloseableHttpResponse response = getResponseOfHttpPost(getUsersPath(), jsonRequest,
+                getHeadersWithBearerToken(token))) {
+            Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpServletResponse.SC_CREATED,
+                    "User creation failed");
+            JSONObject jsonResponse = getJSONObject(EntityUtils.toString(response.getEntity()));
+            return jsonResponse.get("id").toString();
+        } catch (Exception e) {
+            Assert.fail("Error occurred while creating user with bearer token.", e);
+            return null;
+        }
+    }
+
+    /**
+     * Create a user with admin credentials and return the full response as a JSONObject.
+     * This method does not assert the response status code, allowing the caller to handle different scenarios.
+     *
+     * @param adminRegisteredUserInfo UserObject containing user details.
+     * @return JSONObject containing the full response, including status code and any error messages.
+     */
+    public JSONObject createUserAndReturnResponse(UserObject adminRegisteredUserInfo) {
```

Comment:
Why can't we use the `attemptUserCreation()` method,
https://github.com/wso2/product-is/pull/25557/files#diff-495c11c3c4983dce5cb07a1f1f93935546f5d5d27a5cd4fcdfcccc428d7e70bdR844:~:text=public%20CreateUserResponse-,attemptUserCreation,-(UserObject%20userInfo